server/api/api/endpoints/projects_v2.py

```diff
@@ -0,0 +1,176 @@
+
+    background_task = await run_in_threadpool(
+        _get_or_create_project_deletion_background_task,
+        name,
+        deletion_strategy,
+        background_tasks,
+        db_session,
+        auth_info,
+    )
+    response.status_code = http.HTTPStatus.ACCEPTED.value
+    return background_task
+
+
+def _get_or_create_project_deletion_background_task(
+    project_name: str, deletion_strategy: str, background_tasks, db_session, auth_info
+) -> typing.Optional[mlrun.common.schemas.BackgroundTask]:
+    """
+    This method is responsible for creating a background task for deleting a project.
+    The project deletion flow is as follows:
+        When MLRun is leader:
+        1. Create a background task for deleting the project
+        2. The background task will delete the project resources and then project itself
+        When MLRun is a follower:
+        Due to the nature of the project deletion flow, we need to wrap the task as:
+            MLRunDeletionWrapperTask(LeaderDeletionJob(MLRunDeletionTask))
+        1. Create MLRunDeletionWrapperTask
+        2. MLRunDeletionWrapperTask will send a request to the projects leader to delete the project
+        3. MLRunDeletionWrapperTask will wait for the project to be deleted using LeaderDeletionJob job id
+           During (In leader):
+           1. Create LeaderDeletionJob
+           2. LeaderDeletionJob will send a second delete project request to the follower
+           3. LeaderDeletionJob will wait for the project to be deleted using the MLRunDeletionTask task id
+              During (Back here in follower):
+              1. Create MLRunDeletionTask
+              2. MLRunDeletionTask will delete the project resources and then project itself.
+              3. Finish MLRunDeletionTask
+           4. Finish LeaderDeletionJob
+        4. Finish MLRunDeletionWrapperTask
+    """
+    # If the request is from the leader, or MLRun is the leader, we create a background task for deleting the
+    # project. Otherwise, we create a wrapper background task for deletion of the project.
+    background_task_kind_format = (
+        server.api.utils.background_tasks.BackgroundTaskKinds.project_deletion_wrapper
+    )
+    if (
+        server.api.utils.helpers.is_request_from_leader(auth_info.projects_role)
+        or mlrun.mlconf.httpdb.projects.leader == "mlrun"
+    ):
+        background_task_kind_format = (
+            server.api.utils.background_tasks.BackgroundTaskKinds.project_deletion
+        )
+
+    background_task_kind = background_task_kind_format.format(project_name)
+    try:
+        return server.api.utils.background_tasks.InternalBackgroundTasksHandler().get_active_background_task_by_kind(
+            background_task_kind,
+            raise_on_not_found=True,
+        )
+    except mlrun.errors.MLRunNotFoundError:
+        return server.api.utils.background_tasks.InternalBackgroundTasksHandler().create_background_task(
```

Comment:
think its better not to nest under try except because it chains the exception.
see `get_or_create_project` for reference